sdk/dotnet/Pulumi/Deployment/Deployment_RootResource.cs

```diff
@@ -39,12 +39,12 @@ public partial class Deployment
         private async Task<string> SetRootResourceWorkerAsync(Stack stack)
         {
             var resUrn = await stack.Urn.GetValueAsync().ConfigureAwait(false);
-            await this.Engine.SetRootResourceAsync(new SetRootResourceRequest
+            await this.Monitor.SetRootResourceAsync(new SetRootResourceRequest
```

Comment:
i'm not a huge fan of directly exposing the monitor.  can we not go through a fake engine which itself can go through its monitor?

Reply:
ignore the above.

Reply:
I am still trying to reconcile this.  i don't love the incongruity here.  Our system has an 'Engine' and a 'Monitor' for different purposes.  It reads very weird to me to see this as normally a ResourceMonitorClient would not be setting the root resource.

Reply:
Alright... is it a matter of finding a better name? Or do we have to keep two separate entities and mock both?

Reply:
Either `IDispatcher` or `IRpcDispatcher`.  I actually like the latter as it strongly conveys "this is where call out to some other component elsewhere".  It also has good impl names.  i.e. GrpcDispatcher and MockRpcDispatcher.  Both very clear in terms of what they're responsible for.